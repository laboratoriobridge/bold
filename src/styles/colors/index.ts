import { Color } from '../theme/createTheme'

export * from './blue'
export * from './gray'
export * from './green'
export * from './orange'
export * from './purple'
export * from './red'
export * from './turquoise'
export * from './yellow'
export * from './pink'

export interface ColorScale {
  c10: Color
  c20: Color
  c30: Color
  c40: Color
  c50: Color
  c60: Color
  c70: Color
  c80: Color
  c90: Color
  c100: Color
}
