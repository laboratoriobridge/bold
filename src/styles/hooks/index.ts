export * from './useTheme'
export * from './useCss'
export * from './useStyles'
