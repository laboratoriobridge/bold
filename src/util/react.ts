import { MutableRefObject, Ref } from 'react'

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}
