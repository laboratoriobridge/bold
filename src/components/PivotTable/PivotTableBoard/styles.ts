import { css } from 'emotion'
import { Theme } from '../../../styles'
import { blue } from '../../../styles/colors'

export const pivotTableBoardCreateStyles = (theme: Theme) => ({
  tag: css`
    background-color: ${blue.c90};
    color: ${blue.c40};
    border: solid 1px ${blue.c40};
    margin-left: 0.5rem;
    margin-bottom: 0.25rem;
  `,
  tagsContainer: css`
    display: flex;
    flex-wrap: wrap;
  `,
  aggregatorsContainer: css`
    padding: 0.75rem;
    margin: 0.25rem;
    min-height: 7.18rem;
  `,
  filtersContainer: css`
    border: 1px solid ${theme.pallete.gray.c80};
    padding: 0.75rem 1.25rem;
  `,
})
