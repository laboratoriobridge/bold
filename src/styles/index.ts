export * from './hooks'
export { createPallete, Pallete, TextColor } from './theme/createPallete'
export { createTheme, focusBoxShadow, Theme } from './theme/createTheme'
export { createTypography, Typography } from './theme/createTypography'
export { GlobalCss, DefaultGlobalCss } from './global/GlobalCss'
export { ThemeProvider, ThemeProviderProps } from './theme/ThemeContext'
export * from './utils'
export { Styles, withStyles, WithStylesProps } from './withStyles'

import * as palletes from './colors'
export const colors = palletes
