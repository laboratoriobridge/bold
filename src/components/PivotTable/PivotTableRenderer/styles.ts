import { CSSProperties } from 'react'

export const createPivotTableRenderStyles = () => ({
  tableContainer: {
    position: 'relative',
    height: '100%',
  } as CSSProperties,
  tableWrapper: {
    maxWidth: '100%',
    overflow: 'auto',
    display: 'grid',
    placeItems: 'center center',
    placeContent: 'start start',
  } as CSSProperties,
  leftShadow: {
    position: 'absolute',
    left: '0',
    top: '0',
    height: 'calc(100% - 16px)',
    width: '0.5rem',
    background: 'linear-gradient(90deg, rgba(0, 0, 0, 0.12) 10%, rgba(255, 255, 255, 0) 100%)',
  } as CSSProperties,
  rightShadow: {
    position: 'absolute',
    right: '0',
    top: '0',
    height: 'calc(100% - 16px)',
    width: '0.5rem',
    background: 'linear-gradient(270deg, rgba(0, 0, 0, 0.12) 10%, rgba(255, 255, 255, 0) 100%)',
  } as CSSProperties,
})
