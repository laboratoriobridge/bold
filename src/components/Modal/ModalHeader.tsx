import React from 'react'
import { Theme, useStyles } from '../../styles'
import { Heading } from '../Heading'
import { IconImage } from '../Icon'
import { Icon, IconColor } from '../Icon/Icon'
import { HFlow } from '../HFlow'
import { VFlow } from '../VFlow'
import { ModalCloseButton } from './ModalCloseButton'

export type HeaderIconObject = {
  icon: IconImage
  fill?: IconColor
  stroke?: IconColor
}

export type HeaderIconType = IconImage | HeaderIconObject

export interface ModalHeaderProps {
  title: string
  subtitle?: string
  icon?: HeaderIconType
  background?: string
  hasCloseIcon?: boolean
  hasBorder?: boolean
  onCloseButtonClick?: () => void
}

export const ModalHeader = (props: ModalHeaderProps) => {
  const { title, subtitle, icon, background, hasBorder = true, hasCloseIcon = true, onCloseButtonClick } = props

  const { classes } = useStyles(createStyles, background, hasBorder)

  const renderIcon = () => {
    if (typeof icon === 'string') {
      return <Icon icon={icon} size={3} />
    } else {
      const { icon: iconName, fill, stroke } = icon as HeaderIconObject
      return <Icon icon={iconName} size={3} fill={fill} stroke={stroke} />
    }
  }

  return (
    <HFlow
      hSpacing={1}
      justifyContent='space-between'
      alignItems='flex-start'
      style={classes.header}
      data-testid='modal-header'
    >
      <HFlow
        hSpacing={1.25}
        justifyContent='flex-start'
        alignItems={subtitle ? 'flex-start' : 'center'}
        data-testid='modal-header-title-area'
      >
        {icon && renderIcon()}
        <VFlow vSpacing={0}>
          <Heading level={1} color='normal' fontWeight='bold'>
            {title}
          </Heading>
          {subtitle && (
            <Heading level={5} color='normal' fontWeight='normal'>
              {subtitle}
            </Heading>
          )}
        </VFlow>
      </HFlow>
      {hasCloseIcon && <ModalCloseButton onClick={onCloseButtonClick} />}
    </HFlow>
  )
}

export const createStyles = (theme: Theme, background: string, hasDivider: boolean) => ({
  header: {
    position: 'relative',
    zIndex: 1,
    width: '100%',
    height: 'fit-content',
    padding: '1.5rem 1rem 1rem 2rem',
    backgroundColor: background ? background : theme.pallete.surface.main,
    ...(hasDivider && {
      boxShadow: `0 1px 5px 0 ${theme.pallete.divider}, 0 2px 1px -1px ${theme.pallete.divider}`,
    }),
  } as React.CSSProperties,
})
