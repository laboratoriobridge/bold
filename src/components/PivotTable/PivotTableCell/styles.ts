import { FontWeightProperty } from 'csstype'
import { Theme } from '../../../styles'
import { calculateCellColor } from './utils'
import { PivotTableCellProps } from './PivotTableCell'
import { PivotTableCellType } from './model'

export const pivotTableCellCreateStyles = (
  theme: Theme,
  key: string,
  maxValue: number,
  { types, children }: PivotTableCellProps
) => {
  const isValue = types.has(PivotTableCellType.VALUE)
  const isValueOrEmpty = isValue || types.has(PivotTableCellType.EMPTY)
  const isHeaderOrTotal = types.has(PivotTableCellType.HEADER) || types.has(PivotTableCellType.TOTAL)
  const isOnlyValue = isValue && types.size === 1

  const { color, backgroundColor } = calculateCellColor(theme, isOnlyValue, maxValue, children)

  return {
    root: {
      gridArea: key,
      backgroundColor,
      color,
      borderTop: `1px solid ${theme.pallete.divider}`,
      borderLeft: `1px solid ${theme.pallete.divider}`,
      display: 'flex',
      justifyContent: isValueOrEmpty ? 'flex-end' : 'flex-start',
      fontWeight: (isHeaderOrTotal ? 'bold' : 'normal') as FontWeightProperty,
      alignItems: 'center',
      width: '100%',
      height: '100%',
      padding: '0.5rem 1rem',
    },
    endRowBorder: {
      borderBottom: `1px solid ${theme.pallete.divider}`,
    },
    endColumnBorder: {
      borderRight: `1px solid ${theme.pallete.divider}`,
    },
  }
}
