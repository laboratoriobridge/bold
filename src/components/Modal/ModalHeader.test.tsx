import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { createTheme, ThemeContext } from '../../styles'
import { HeaderIconObject, ModalHeader } from './ModalHeader'

describe('modal header component', () => {
  describe('basic rendering', () => {
    it('should render correctly', () => {
      const { container } = render(<ModalHeader title='title' />)
      expect(container).toMatchSnapshot()
    })

    it('should render the title when "title" prop is provided', () => {
      render(<ModalHeader title='title' />)
      expect(screen.getByText('title')).toBeInTheDocument()
    })

    it('should render the subtitle when "subtitle" prop is provided', () => {
      render(<ModalHeader title='title' subtitle='subtitle' />)
      expect(screen.getByText('subtitle')).toBeInTheDocument()
    })

    it('should render only the title when only "title" prop is provided', () => {
      render(<ModalHeader title='title' />)
      const texts = screen.getAllByText('title')
      expect(texts).toHaveLength(1)
      expect(screen.getByTestId('modal-header').textContent).toBe('title')
    })
  })

  describe('icon', () => {
    it('should render the icon when "icon" prop is provided as IconImage', () => {
      render(<ModalHeader title='title' hasCloseIcon={false} icon='infoCircleOutline' />)
      expect(screen.getByTestId('modal-header').querySelector('svg')).toBeInTheDocument()
    })

    it('should render the icon when "icon" prop is provided as HeaderIconObject', () => {
      const theme = createTheme()
      const iconObj: HeaderIconObject = {
        icon: 'infoCircleOutline',
        fill: 'primary',
        stroke: 'danger',
      }

      render(
        <ThemeContext.Provider value={theme}>
          <ModalHeader title='title' hasCloseIcon={false} icon={iconObj} />
        </ThemeContext.Provider>
      )

      const svg = screen.getByTestId('modal-header').querySelector('svg')
      expect(svg).toBeInTheDocument()
      expect(svg).toHaveStyle(`fill: ${theme.pallete.primary.main};`)
      expect(svg).toHaveStyle(`stroke: ${theme.pallete.status.danger.main};`)
    })

    it('should not render svg icon when icon prop is not provided', () => {
      render(<ModalHeader title='title' hasCloseIcon={false} />)
      expect(screen.getByTestId('modal-header').querySelector('svg')).toBeNull()
    })
  })

  describe('title area alignment', () => {
    it('should set align-items to flex-start when subtitle is provided', () => {
      render(<ModalHeader title='title' subtitle='subtitle' />)
      const container = screen.getByTestId('modal-header-title-area')
      expect(container).toHaveStyle('align-items: flex-start')
    })

    it('should set align-items to center when subtitle is not provided', () => {
      render(<ModalHeader title='title' />)
      const container = screen.getByTestId('modal-header-title-area')
      expect(container).toHaveStyle('align-items: center')
    })
  })

  describe('styles', () => {
    const theme = createTheme()

    it('should apply background-color when "backgroundColor" prop is provided', () => {
      render(
        <ThemeContext.Provider value={theme}>
          <ModalHeader title='title' backgroundColor='red' />
        </ThemeContext.Provider>
      )
      expect(screen.getByTestId('modal-header')).toHaveStyle(`background-color: red`)
    })

    it('should apply background-color default when "backgroundColor" prop is not provided', () => {
      render(
        <ThemeContext.Provider value={theme}>
          <ModalHeader title='title' />
        </ThemeContext.Provider>
      )
      expect(screen.getByTestId('modal-header')).toHaveStyle(`background-color: ${theme.pallete.surface.main}`)
    })

    it('should apply box-shadow when "hasDivider" is true', () => {
      render(
        <ThemeContext.Provider value={theme}>
          <ModalHeader title='title' hasDivider={true} />
        </ThemeContext.Provider>
      )
      expect(screen.getByTestId('modal-header')).toHaveStyle(
        `box-shadow: 0 1px 5px 0 ${theme.pallete.divider},0 2px 1px -1px ${theme.pallete.divider}`
      )
    })

    it('should apply box-shadow when "hasDivider" is not provided (default true)', () => {
      render(
        <ThemeContext.Provider value={theme}>
          <ModalHeader title='title' />
        </ThemeContext.Provider>
      )
      expect(screen.getByTestId('modal-header')).toHaveStyle(
        `box-shadow: 0 1px 5px 0 ${theme.pallete.divider},0 2px 1px -1px ${theme.pallete.divider}`
      )
    })

    it('should not apply box-shadow when "hasDivider" is false', () => {
      render(<ModalHeader title='title' hasDivider={false} />)
      expect(getComputedStyle(screen.getByTestId('modal-header')).boxShadow).toBe('')
    })
  })

  describe('close button', () => {
    it('should render close button when "hasCloseIcon" is true', () => {
      render(<ModalHeader title='title' hasCloseIcon={true} />)
      expect(screen.getByRole('button')).toBeInTheDocument()
    })

    it('should render close button when "hasCloseIcon" is not provided (default true)', () => {
      render(<ModalHeader title='title' />)
      expect(screen.getByRole('button')).toBeInTheDocument()
    })

    it('should not render close button when "hasCloseIcon" is false', () => {
      render(<ModalHeader title='title' hasCloseIcon={false} />)
      expect(screen.queryByRole('button')).toBeNull()
    })

    it('should call onCloseButtonClick when close button is clicked and "hasCloseIcon" is true', () => {
      const onCloseMock = jest.fn()
      render(<ModalHeader title='title' hasCloseIcon={true} onCloseButtonClick={onCloseMock} />)
      fireEvent.click(screen.getByRole('button'))
      expect(onCloseMock).toHaveBeenCalledTimes(1)
    })

    it('should call onCloseButtonClick when close button is clicked and "hasCloseIcon" is not provided (default true)', () => {
      const onCloseMock = jest.fn()
      render(<ModalHeader title='title' onCloseButtonClick={onCloseMock} />)
      fireEvent.click(screen.getByRole('button'))
      expect(onCloseMock).toHaveBeenCalledTimes(1)
    })

    it('should not call onCloseButtonClick when "hasCloseIcon" is false', () => {
      const onCloseMock = jest.fn()
      render(<ModalHeader title='title' hasCloseIcon={false} onCloseButtonClick={onCloseMock} />)
      expect(screen.queryByRole('button')).toBeNull()
      expect(onCloseMock).not.toHaveBeenCalled()
    })
  })
})
