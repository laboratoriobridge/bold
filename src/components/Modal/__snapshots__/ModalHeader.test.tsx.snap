// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader basic rendering should render correctly with children 1`] = `
.emotion-2 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: minmax(min-content,max-content);
  grid-gap: 0.5rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  grid-template-columns: 1fr auto;
  padding: 1.5rem 1rem 1rem 2rem;
  z-index: 1;
  background-color: #FFFFFF;
  border-bottom: none;
  grid-row: 1;
  grid-column: 1 / -1;
}

.emotion-1 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
}

.emotion-1 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-1:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

<div>
  <div
    class="emotion-2"
    data-testid="modal-header"
  >
    <div>
      Custom header
    </div>
    <button
      aria-label="Close"
      class="emotion-1"
      title="Close"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-0"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0v.001z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`ModalHeader basic rendering should render correctly with props 1`] = `
.emotion-8 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: minmax(min-content,max-content);
  grid-gap: 0.5rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  grid-template-columns: 1fr auto;
  padding: 1.5rem 1rem 1rem 2rem;
  z-index: 1;
  background-color: #FFFFFF;
  border-bottom: none;
  grid-row: 1;
  grid-column: 1 / -1;
}

.emotion-5 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: minmax(min-content,max-content);
  grid-gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  display: inline-block;
  margin-bottom: 0rem;
  margin-top: 0rem;
}

.emotion-1:first-of-type {
  margin-top: 0;
}

.emotion-1:last-of-type {
  margin-bottom: 0;
}

.emotion-1:empty {
  display: none;
}

.emotion-0 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 1.5rem;
  line-height: 1.5;
  color: #24252E;
  font-weight: bold;
  color: #24252E;
  font-weight: bold;
}

.emotion-2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #24252E;
  font-weight: bold;
  color: #24252E;
  font-weight: normal;
}

.emotion-7 {
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5rem;
  color: #24252E;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #24252E;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
}

.emotion-7 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-7:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-7:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0069D0;
}

.emotion-7:not(:disabled):hover {
  background-color: rgba(58,58,71,0.1);
}

.emotion-6 {
  fill: currentColor;
  font-size: 1.5rem;
}

<div>
  <div
    class="emotion-8"
    data-testid="modal-header"
  >
    <div
      class="emotion-5"
    >
      <div
        icon="infoCircleFilled"
      />
      <div
        class="emotion-4"
      >
        <div
          class="emotion-1"
        >
          <h1
            class="emotion-0"
          >
            title
          </h1>
        </div>
        <div
          class="emotion-1"
        >
          <h5
            class="emotion-2"
          >
            subtitle
          </h5>
        </div>
      </div>
    </div>
    <button
      aria-label="Close"
      class="emotion-7"
      title="Close"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="emotion-6"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            clip-rule="evenodd"
            d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0v.001z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
